arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/RaiseRacingTest.kt

```diff
@@ -147,29 +147,39 @@ class RaiseRacingTest {
     assertFailsWith<IllegalStateException> {
       either<String, String> {
         racing {
-          raceOrThrow { throw IllegalStateException("Test exception") }
+          raceOrFail { throw IllegalStateException("Test exception") }
           race { awaitCancellation() }
         }
       }
     }
   }
 
+  /* Note on the implementation:
+   * without the outer try/catch the test fails
+   * because [runTest] overrides the default coroutine
+   * exception handler, conflicting with our own usage.
```

Comment:
I think you just need to pass in a coroutine exception handler into the `race` calls, no? Or use `raceOrFail`